public/friendlycode/js/fc/load-project-files.js

```diff
@@ -0,0 +1,168 @@
+  var self = {};
+
+  // Taken from http://stackoverflow.com/questions/6965107/converting-between-strings-and-arraybuffers
+  function convertToBuffer(string) {
+    var buf = new ArrayBuffer(string.length);
+    var bufView = new Uint8Array(buf);
+    for(var i = 0, strLen = string.length; i < strLen; i++) {
+      bufView[i] = string.charCodeAt(i);
+    }
+    return buf;
+  }
+
+  self.load = function(project, defaultTemplate, callback) {
+    var fs = Bramble.getFileSystem();
+    var shell = new fs.Shell();
+    var projectFilesUrl = window.location.protocol + "//" + window.location.host + "/initializeProject";
+    var request = new XMLHttpRequest();
+    var root = project && project.title;
+    var defaultProject = false;
+    var files;
+
+    if(!root) {
+      callback(new Error("No project specified"));
+      return;
+    }
+
+    root = Path.join("/", root);
+
+    if(typeof defaultTemplate !== "function") {
+      defaultProject = true;
+      project.dateCreated = (new Date()).toISOString();
+      project.dateUpdated = project.dateCreated;
+      files = self.generateDefaultFiles(defaultTemplate);
+      updateFs();
+      return;
+    }
+
+    callback = defaultTemplate;
+
+    function relative(path) {
+      if(!Path.isAbsolute(path)) {
+        return path;
+      }
+
+      var temp = path;
+      var relPath = "";
+      var exit = false;
+
+      while(!exit) {
+        if(temp === root) {
+          exit = true;
+        } else {
+          relPath = "/" + Path.basename(temp) + relPath;
+        }
+
+        temp = Path.dirname(temp);
+      }
+
+      return relPath.substr(1);
+    }
+
+    function updateFs() {
+      if(!defaultProject && request.readyState !== 4) {
+        return;
+      }
+
+      if(!defaultProject && request.status !== 200) {
+        // TODO: handle error case here
+        callback(new Error("Failed to get files for this project"));
+        return;
+      }
+
+      if(!files) {
+        try {
+          files = request.response.files;
+        } catch(e) {
+          // TODO: handle error case here
+          callback(new Error("Failed to get a response"));
+          return;
+        }
+      }
+
+      var length = files.length;
+      var completed = 0;
+      var filePathToOpen;
+
+      function checkProjectLoaded() {
+        if(completed === length) {
+          callback(null, {
+            root: root,
+            open: filePathToOpen
+          });
+        }
+      }
+
+      function writeFile(path, data) {
+        var parent = Path.dirname(path);
+
+        function write() {
+          fs.writeFile(path, data, function(err) {
+            if(err) {
+              // TODO: handle error case here
+              console.error("Failed to write: ", path);
+              callback(err);
+              return;
+            }
+
+            completed++;
+            checkProjectLoaded();
+          });
+        }
+
+        fs.stat(parent, function(err) {
```

Comment:
Don't bother with the `stat` call, just do `mkdirp` always, and if it's there already, don't fail.